x-pack/solutions/observability/plugins/profiling/public/views/storage_explorer/index.tsx

```diff
@@ -59,10 +64,32 @@ export function StorageExplorerView() {
     ]
   );
 
+  if (!loadedState.summary && storageExplorerSummaryState.status === AsyncStatus.Settled) {
+    setLoadedState((prevState) => ({ ...prevState, summary: true }));
+  }
```

Comment:
We need this to be in a `useEffect` to prevent re-renders

Reply:
I originally left out `useEffect` here because I didn't see a difference between the number of times it was called. This was probably due to the fact I was over-capturing with the dependency list. Thanks for the good feedback, this has been enlightening.